File: POLEVAL.lua
local i = 0
while true do
    local i, n, c, x = i + 1, io.read("*n"), {}, {}
    if n == -1 then
        break
    end
    for j = 0, n do
        c[j] = io.read("*n")
    end
    local k = io.read("*n")
    for j = 1, k do
        x[j] = io.read("*n")
    end
    print(string.format("Case %d:", i))
    for j = 1, k do
        local r, p = c[0], x[j]
        for k = 1, n do
            r = r * p + c[k]
        end
        print(r)
    end
end

File: POLEVAL.py
i = 0
while True:
    i += 1
    n = int(input())
    if n == -1:
        break
    c = tuple(map(int, reversed(input().split())))
    k = int(input())
    x = tuple(map(int, input().split()))
    print("Case {}:".format(i))
    for j in x:
        r = c[0]
        for k in range(1, n + 1):
            r += c[k] * pow(j, k)
        print(r)
